src/main/java/org/testng/TestNG.java

```diff
@@ -960,7 +991,10 @@ private void addListeners(XmlSuite s) {
             "Listener " + listenerName + " was not found in project's classpath");
       }
 
-      addListener((ITestNGListener) m_objectFactory.newInstance(listenerClass));
+      BasicAttributes basic = new BasicAttributes(null, listenerClass);
```

Comment:
I don't catch the reason for this change: the dispenser API is more difficult and `m_objectFactory.newInstance` is already supposed to manage guice.

From a design point of view, we have to choose between the `ObjectFactory` API and `ObjectDispenser` API because it looks like both has the same responsibilities.

Reply:
The `Dispenser` has the intelligence to figure out what specific dispenser flavor to use based on the `@Guice` annotation and it does this using an embedded `IObjectFactory`. The `IObjectFactory` and the dispenser api dont share the same responsibilities. 

The `org.testng.internal.objects.Dispenser` that produces instances of `IObjectDispenser` types uses the `IObjectFactory` only for the `SimpleObjectDispenser` (this uses the `IObjectFactory` to instantiate objects)

Reply:
Sure but they have the same objective: creating objects.
Maybe ObjectFactory could be the API level of the internal Dispenser.

But we should not mix the usage of both.

Reply:
Sure. But the way I look at is, Dispenser is the higher level API that abstracts out the object creation part and it internally leverages the `IObjectFactory` implementation for object creation when we dont involve Dependency Injection via `Guice`